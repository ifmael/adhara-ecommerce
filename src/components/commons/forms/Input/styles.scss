@import "styles/_variables.scss";

.input-container {
  margin-bottom: 1.5rem;

  .input-label-container {
    display: flex;
    margin-bottom: 0.5rem;
  }

  .input-required {
    color: $generalLink;
    margin-left: auto;
  }

  .input-field {
    display: flex;
    align-items: center;
    border: 0.1rem solid $generalLink;
    border-radius: 2rem;
    color: $generalLink;
  }

  .input-field {
    height: 4rem;
  }

  .single-icon {
    font-size: 2rem;
    margin-left: 1.1rem;
  }

  .input-field-value {
    color: $generalLink;
    font-size: 1.6rem;
    margin: 1rem;
    width: 100%;
    border: none;

    &:focus {
      outline-width: 0;
    }

    &::placeholder {
      color: $generalLink;
    }
  }
}
