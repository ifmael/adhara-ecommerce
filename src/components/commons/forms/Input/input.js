import React from "react";
import { bool, func, string } from "prop-types";
import "./styles.scss";

const propTypes = {
  name: string,
  onChangeHandler: func,
  placeHolder: string,
  required: bool,
  text: string,
  type: string,
  value: string,
};

const defaultProps = {
  name: "",
  onChangeHandler: () => {},
  placeHolder: "",
  required: false,
  text: "",
  type: "text",
  value: "",
};

const Input = ({
  name,
  onChangeHandler,
  placeHolder,
  required,
  text,
  type,
  value,
}) => {
  const labelInput = (
    <div className="input-label-container">
      <label name={name} className="input-label" htmlFor={name}>
        {text}
        {required ? "*" : ""}
      </label>
    </div>
  );

  return (
    <div className="input-container">
      {text ? labelInput : null}
      <div className="input-field">
        <input
          id={name}
          className="input-field-value"
          name={name}
          onChange={(event) => onChangeHandler(event.target.value, name)}
          placeholder={placeHolder}
          value={value}
          type={type}
        />
      </div>
    </div>
  );
};

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;
