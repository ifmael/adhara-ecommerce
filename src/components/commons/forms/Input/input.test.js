import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Input from "./index.js";

describe("Input test", () => {
  test("take a snapshot", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment(<Input />)).toMatchSnapshot();
  });

  test("Check basic properties", () => {
    render(
      <Input placeHolder="A placeholder" text="Test 01" value="input value" />
    );

    expect(screen.getByText(/Test 01/i)).toHaveTextContent("Test 01");
    const { placeholder, type, value } = screen.getByRole("textbox");
    expect(placeholder).toBe("A placeholder");
    expect(type).toBe("text");
    expect(value).toBe("input value");
  });

  test("Check onChange function", async () => {
    const handler = jest.fn((string) => (value = string));
    let value = "Initial value";

    render(<Input onChangeHandler={handler} value={value} />);
    await userEvent.type(screen.getByRole("textbox"), "Hello, World!");
    expect(handler.mock.calls.length).toBe(13);
  });
});
