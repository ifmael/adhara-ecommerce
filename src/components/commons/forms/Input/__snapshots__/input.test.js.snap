// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="input-container"
  >
    <div
      class="input-field"
    >
      <input
        class="input-field-value"
        id=""
        name=""
        placeholder=""
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
